internal/client/client.go

```diff
@@ -105,10 +103,22 @@ type Client struct {
 	PartitionService     *icluster.PartitionService
 	ClusterService       *icluster.Service
 	name                 string
-	shutdownHandlers     []ShutdownHandler
+	ShutdownHandlers     []func(ctx context.Context)
 	state                int32
 }
 
+func (c *Client) AddShutdownHandler(handler func(ctx context.Context)) {
+	// this is supposed to be called during client initialization, so there's no risk of races.
+	c.ShutdownHandlers = append(c.ShutdownHandlers, handler)
+}
+
+func (c *Client) ExecuteShutdownHandlers(ctx context.Context) {
```

Comment:
Is this method required? Do we expect to call it from somewhere other than the internal client?

Reply:
I don't think we expect to call it. I inlined it to where it is being called. [17f8ebc](https://github.com/hazelcast/hazelcast-go-client/pull/772/commits/17f8ebc0e27f5a5d71baf61a7eb5b9f5253c25bf).